packages/flutter/lib/src/material/material_app.dart

```diff
@@ -73,6 +73,8 @@ class _MaterialAppState extends State<MaterialApp> implements BindingObserver {
   GlobalObjectKey _navigator;
 
   Size _size;
+  double _devicePixelRatio;
```

Comment:
oh we should definitely do the same for all of these, yes. My comment was about the existing approach, not your specific changes, except insofar as your changes exacerbate the problem with the existing approach. :-)